assets/js/theme/common/carousel/utils/playPause.js

```diff
@@ -0,0 +1,38 @@
+import { throttle } from 'lodash';
+
+const playAction = 'slickPlay';
+const pauseAction = 'slickPause';
+
+export default ($heroCarousel) => {
+    const $playPauseButton = $('.js-hero-play-pause-button');
+
+    if ($playPauseButton.length === 0) return;
+
+    const slickSettings = $heroCarousel[0].slick;
+    if (!slickSettings) return;
+
+    const { slideCount } = slickSettings;
+    if (slideCount < 2) {
+        $playPauseButton.css('display', 'none');
+        return;
+    }
+
+    const onPlayPauseClick = () => {
+        const isPlay = $playPauseButton.data('play');
+        const action = isPlay ? pauseAction : playAction;
+        const {
+            play,
+            ariaPlay,
+            pause,
+            ariaPause,
+        } = $playPauseButton.data('labels');
+
+        $heroCarousel.slick(action);
+        $playPauseButton
+            .data('play', !isPlay)
+            .text(action === playAction ? pause : play)
+            .attr('aria-label', action === playAction ? ariaPause : ariaPlay);
+    };
+
+    $playPauseButton.on('click', throttle(onPlayPauseClick, 1000, { trailing: false }));
```

Comment:
Could you please explain why we set throttle to 1000 here?

Reply:
To prevent toggle this option very fast by user. It might cause some glithes or performance issues. Slick adds such throttle by default to slider controls like arrows and dots for 300. Maybe we should use the same value instead of 1000?

Reply:
Did you try 300?

Reply:
Our slick version has 500 as default value. Now I am using speed value from slick options to make transition time and throttle equal